liblangutil/Scanner.cpp

```diff
@@ -563,12 +563,14 @@ void Scanner::scanToken()
 				token = Token::Add;
 			break;
 		case '-':
-			// - -- -=
+			// - -- -= ->
 			advance();
 			if (m_char == '-')
 				token = selectToken(Token::Dec);
 			else if (m_char == '=')
 				token = selectToken(Token::AssignSub);
+			else if (m_char == '>' && m_kind == ScannerKind::Yul)
```

Comment:
I actually think `->` is invalid in Solidity too, so could just enable this new token for both modes.

Reply:
Run all the tests with this enabled for Solidity, not a single failure.